### PR TITLE
#2581: add `initialize` trigger

### LINKED ISSUE #2581: Add "initialize" trigger type
Context
- Currently we have an "appear" trigger which triggers when the element comes into the view port
- In many cases we want to run immediately when the DOM element is added so that there's no flicker (e.g., if hiding/highlighting the element)

### PR DESCRIPTION
**Pull request by twschiller**
Closes #2581 

Adds an "initialize" trigger, which fires on DOM attachment vs. appear

### CODE REVIEW

File: src/extensionPoints/triggerExtension.ts [hunk 1]
```diff
@@ -75,8 +75,10 @@ export type Trigger =
   | "interval"
   // `appear` is triggered when an element enters the user's viewport
   | "appear"
-  | "click"
+  // `initialize` is triggered when an element is added to the DOM
+  | "initialize"
```
Comment (fregante) [inline src/extensionPoints/triggerExtension.ts:79]: Any other names? Maybe "add", "insert", "create"?

File: src/extensionPoints/triggerExtension.ts [hunk 2]
```diff
@@ -418,6 +428,37 @@ export abstract class TriggerExtensionPoint extends ExtensionPoint<TriggerConfig
...
+        void this.runTrigger(element);
+      }
+
+      return;
+    }
+
+    this.initializeObserver = initialize(
+      this.triggerSelector,
+      (index, element) => {
+        void this.runTrigger(element as HTMLElement).then((errors) => {
```
Comment (fregante) [inline src/extensionPoints/triggerExtension.ts:449]: Nit: We can use a regular async function here
```suggestion
      async (index, element) => {
        const errors = await this.runTrigger(element as HTMLElement);
```

File: src/extensionPoints/triggerExtension.ts [hunk 3]
```diff
@@ -525,9 +566,14 @@ export abstract class TriggerExtensionPoint extends ExtensionPoint<TriggerConfig
   }
 
   private cancelObservers() {
+    this.intervalController?.abort();
```
Comment (fregante) [inline src/extensionPoints/triggerExtension.ts:569]: Nit: We could use abort controllers for everything so we don't have to have one property for each. For example there can be just one controller register in the constructor and then all the observers attach an `abort` listener to it. Then this function can just be:

```js
private cancelObservers() {
    this.abortController.abort();
}
```

File: src/extensionPoints/triggerExtension.ts [hunk 4]
```diff
@@ -418,6 +428,37 @@ export abstract class TriggerExtensionPoint extends ExtensionPoint<TriggerConfig
...
+        void this.runTrigger(element as HTMLElement).then((errors) => {
+          if (errors.length > 0) {
+            console.error("An error occurred while running a trigger", {
+              errors,
+            });
+            notifyError("An error occurred while running a trigger");
+          }
+        });
+      },
+      { target: document }
```
Comment (fregante) [inline src/extensionPoints/triggerExtension.ts:458]: I think this is the default

File: src/extensionPoints/triggerExtension.ts (hunk 1 above)
Comment (twschiller) [reply src/extensionPoints/triggerExtension.ts:79]: In the configuration I'm going to keep as-is with initialize. That's how JS libraries seem to refer to refer to it

We'll have to change the names of events in the page editor. E.g., non-developers don't know what a "blur" handler is

File: src/extensionPoints/triggerExtension.ts (hunk 3 above)
Comment (twschiller) [reply src/extensionPoints/triggerExtension.ts:569]: Good idea, the current state is getting super messy. Will see what will be involved in a cleanup

Comment (fregante) [reply src/extensionPoints/triggerExtension.ts:569]: On it

File: src/extensionPoints/triggerExtension.ts (hunk 4 above)
Comment (fregante) [reply src/extensionPoints/triggerExtension.ts:458]: Apparently not… :(

File: src/extensionPoints/triggerExtension.ts (hunk 3 above)
Comment (fregante) [reply src/extensionPoints/triggerExtension.ts]: Pushed. This is what it looks like, what do you think?

File: src/extensionPoints/triggerExtension.ts [hunk 5]
```diff
@@ -184,10 +160,10 @@ export abstract class TriggerExtensionPoint extends ExtensionPoint<TriggerConfig
   private readonly boundEventHandler: JQuery.EventHandler<unknown>;
 
   /**
-   * Controller to abort/cancel the currently running interval loop
+   * Controller to drop all listeners and timers
    * @private
    */
-  private intervalController: AbortController | null;
+  private abortController = new AbortController();
```
Comment (fregante) [inline src/extensionPoints/triggerExtension.ts:166]: This is a new JS feature by the way. You can initialize properties right here and they'll be called in the `constructor`. In most cases you can replace:

```js
class {
	  constructor() {
		  this.prop = myInitialization()
	  }
}
```

with:

```js
class {
	prop = myInitialization()
}

Comment (fregante) [reply src/extensionPoints/triggerExtension.ts:166]: This actually allows for some _coool_ usage like pre-bound methods:


```js
class {
   constructor() {
     super(id, name, description, icon);
     // Bind so we can pass as callback
     this.boundEventHandler = this.eventHandler.bind(this);
   }
   eventHandler() {
     // Bind me please
   } 
}
```

becomes:

```js
class {
   eventHandler = () => {
     // Look ma, no constructor()
   }
}
```

**Comment by twschiller** [comment]
Definitely looks cleaner. I changed onCancel to addCancelHandler just to clarify a bit

**Pull request merged**